osu.Game/Skinning/Skin.cs

```diff
@@ -210,5 +215,57 @@ protected virtual void Dispose(bool isDisposing)
         }
 
         #endregion
+
+        public class SquishingTextureLoaderStore : IResourceStore<TextureUpload>
+        {
+            private readonly IResourceStore<TextureUpload> textureStore;
+
+            public SquishingTextureLoaderStore(IResourceStore<TextureUpload> textureStore)
+            {
+                this.textureStore = textureStore;
+            }
+
+            public void Dispose()
+            {
+                textureStore.Dispose();
+            }
+
+            public TextureUpload Get(string name)
+            {
+                var textureUpload = textureStore.Get(name);
+
+                // NRT not enabled on framework side classes (IResourceStore / TextureLoaderStore), welp.
+                if (textureUpload.IsNull())
+                    return null!;
+
+                // So there's a thing where some users have taken it upon themselves to create skin elements of insane dimensions.
+                // To the point where GPUs cannot load the textures (along with most image editor apps).
+                // To work around this, let's look out for any stupid images and shrink them down into a usable size.
+                const int max_supported_texture_size = 16384;
```

Comment:
Didn't we find with the o!f texture atlas PR that 8192 is the most common/widely supported texture size?

Reply:
Oops, I meant to set this to 8192 but was experimenting higher.